### DISCUSSION TITLE
Can we create this using Paste: select values + tags like in Messaging Insights filter

### DISCUSSION THREAD

**Discussion by elainelylee** [twilio-labs/paste#993] [General] [answered]
This is a screenshot of the filter in Messaging Insights. Would like to do the same in the product I'm working on to have some consistency. Mainly, can we Paste-ify the select drowndown that shows the tags that you can easily remove? OR, use the console react component for it and swap it for a Paste version when available?

![image](https://user-images.githubusercontent.com/6718764/100954817-301a4c00-34ca-11eb-812f-2b6df386407b.png)

**Comment by SiTaggart** [comment, marked as answer]
Hi @elainelylee 

The multi select combobox is on the roadmap, but not scheduled until maybe Q2 of next year.

You can add any element to the prefix of a combobox, but we don't have pills yet either so that may not be particularly helpful.

My personal preference would be to just use the react-components one and wait for the new one to drop next year for the swap. Avoid, if you can, restyling the old one to make it look like the new styles as you'll be incurring twice the technical debt.

**Comment by elainelylee** [comment]
Got it. Does this mean we shouldn't update any of those fields in that
grouping using the new Paste select?
Alternatively, only for that "Value" dropdown, use the react-component and
swap that out later.

Elaine Lee
Principal Product Designer - Twilio AI
[image: twilio] <https://www.twilio.com/?utm_source=email_signature>
EMAIL ellee@twilio.com

**Reply by SiTaggart** [reply]
Just the multi select. Use Paste components for everything else in the UI.

**Reply by elainelylee** [reply]
Are we ok with using this same pattern where the labels are removed when you "add filter" row? cc @epardosanllorente

**Reply by SiTaggart** [reply]
It's preferable to show the label on all fields as it gives a much bigger click target for each field (clicking a label should focus it's input).

For this specific case, as the labels are all the same for each row, it's probably fine to visually hide the label of subsequent rows using the [Screen reader only](https://paste.twilio.design/components/screen-reader-only) component, so long as the first row is visually showing their labels.

In the long term you might end up putting them in some sort of interactive / editable data grid where the column headers act as the visual labels too, once we have the data grid available.

**Reply by epardosanllorente** [reply]
So for the input label, we can use the Screen ready only component but for the combobox component, there is no such solution, right? 
If we would like to hide the label for the combobox we will need to implement our own solution without using the combobox component.

**Reply by SiTaggart** [reply]
`labelText` for the Combobox takes children. You can pass it a `<ScreenReaderOnly />` wrapped string as the label to achieve the same functionality.

As an example I've created a CodeSandbox https://codesandbox.io/s/input-and-combobox-with-visually-hidden-labels-3xzum

**Reply by epardosanllorente** [reply]
thank you, it is working 🎉
